pkg/install/install.go

```diff
@@ -0,0 +1,246 @@
+		}
+
+		vsl := new(apiextv1beta1.CustomResourceDefinition)
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredVSL.Object, vsl); err != nil {
+			return false, errors.Wrap(err, "error converting volumesnapshotlocation from unstructured")
+		}
+
+		if !(crdIsReady(*bsl) && crdIsReady(*vsl)) {
+			return false, nil
+		}
+
+		return true, nil
+	})
+	return isReady, err
+}
+
+func isAvailable(c appsv1beta1.DeploymentCondition) bool {
+	// Make sure that the deployment has been available for at least 10 seconds.
+	// This is because the deployment can show as Ready momentarily before the pods fall into a CrashLoopBackOff.
+	// See podutils.IsPodAvailable upstream for similar logic with pods
+	if c.Type == appsv1beta1.DeploymentAvailable && c.Status == corev1.ConditionTrue {
+		if !c.LastTransitionTime.IsZero() && c.LastTransitionTime.Add(10*time.Second).Before(time.Now()) {
+			return true
+		}
+	}
+	return false
+}
+
+func deploymentIsReady(factory client.DynamicFactory) (bool, error) {
+	gvk := schema.FromAPIVersionAndKind(appsv1beta1.SchemeGroupVersion.String(), "Deployment")
+	apiResource := metav1.APIResource{
+		Name:       "deployments",
+		Namespaced: true,
+	}
+	c, err := factory.ClientForGroupVersionResource(gvk.GroupVersion(), apiResource, "velero")
+	if err != nil {
+		return false, errors.Wrapf(err, "Error creating client for deployment polling")
+	}
+	// declare this variable out of scope so we can return it
+	var isReady bool
+	var readyObservations int32
+	err = wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+		unstructuredDeployment, err := c.Get("velero", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return false, nil
+		} else if err != nil {
+			return false, errors.Wrap(err, "error waiting for deployment to be ready")
+		}
+
+		deploy := new(appsv1beta1.Deployment)
+		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredDeployment.Object, deploy); err != nil {
+			return false, errors.Wrap(err, "error converting deployment from unstructured")
+		}
+
+		for _, cond := range deploy.Status.Conditions {
+			if isAvailable(cond) {
+				readyObservations++
+			}
+		}
+		// Make sure we query the deployment enough times to see the state change, provided there is one.
+		if readyObservations > 4 {
+			isReady = true
+		} else {
+			return false, nil
+		}
+		return true, nil
+	})
+	return isReady, err
+}
+
+// GroupResources groups resources into ResourcesGroups based on whether the resources are CustomResourceDefinitions or other types of kubernetes objects
+// This is useful to wait for readiness before creating CRD objects
+func GroupResources(resources *unstructured.UnstructuredList) []*ResourceGroup {
+	crdObjs := new(ResourceGroup)
+	crdObjs.Ready = crdsAreReady
+	crdObjs.UnreadyMessage = "CRDs were not fully ready to create objects"
+
+	otherObjs := new(ResourceGroup)
```

Comment:
It would be nice to wait for the restic daemonset to be "ready", too, if it's being installed - have to think about what that would look like.